crates/kitsune_p2p/kitsune_p2p/src/gossip/simple_bloom/step_4_com_loop_inner.rs

```diff
@@ -0,0 +1,337 @@
+use super::*;
+use kitsune_p2p_types::codec::*;
+
+pub(crate) async fn step_4_com_loop_inner_outgoing(
+    inner: &Share<SimpleBloomModInner>,
+    tuning_params: KitsuneP2pTuningParams,
+    space: Arc<KitsuneSpace>,
+    ep_hnd: Tx2EpHnd<wire::Wire>,
+    peer_cert: Tx2Cert,
+    how: HowToConnect,
+    gossip: GossipWire,
+) -> KitsuneResult<()> {
+    let gossip = gossip.encode_vec().map_err(KitsuneError::other)?;
+    let gossip = wire::Wire::gossip(space, gossip.into());
+
+    let t = tuning_params.implicit_timeout();
+
+    let con = match how {
+        HowToConnect::Con(con) => {
+            if con.is_closed() {
+                let url = pick_url_for_cert(inner, &peer_cert)?;
+                ep_hnd.get_connection(url, t).await?
+            } else {
+                con
+            }
+        }
+        HowToConnect::Url(url) => ep_hnd.get_connection(url, t).await?,
+    };
+    con.notify(&gossip, t).await?;
+
+    Ok(())
+}
+
+pub(crate) async fn step_4_com_loop_inner_incoming(
+    inner: &Share<SimpleBloomModInner>,
+    con: Tx2ConHnd<wire::Wire>,
+    gossip: GossipWire,
+) -> KitsuneResult<()> {
+    use crate::event::*;
+
+    // parse the message
+    let (send_accept, remote_filter) = match gossip {
+        GossipWire::Initiate(Initiate { filter }) => {
+            let bloom_byte_count = filter.len();
+            tracing::debug!(
+                %bloom_byte_count,
+                "incoming 'Initiate'",
+            );
+
+            (true, filter)
+        }
+        GossipWire::Accept(Accept { filter }) => {
```

Comment:
Does this mean we:
1. initiate -> filter 
2. accept <- filter
3. chunk -> data
Why not just send the missing data on step two and save the network message? Or am I misunderstanding this?

Reply:
This is the bloom filter exchange - The initiator sends their filter to a target through the "initiate" message. The acceptor sends their filter to the initiator through the "accept" message. Then both sides build chunks given the filters to send to the other side.